### PR TITLE
cpeng: change alignment to xfer limit

### PR DESCRIPTION
**Pull request by r-rojo**
data request limit is the PCIe transfer width that the copy engine IP
will use. It can be 64, 128, 512, or 1024 bytes.
This change ensures that chunk sizes are aligned with data request
limit.

Signed-off-by: Rodrigo Rojo <rodrigo.rojo@intel.com>

### CODE REVIEW

File: ofs/apps/cpeng/hps.cpp [hunk 1]
```diff
@@ -92,6 +104,10 @@ class cpeng : public opae::afu_test::command
       ->transform(CLI::AsNumberWithUnit(units));
     app->add_option("-c,--chunk", chunk_, "Chunk size. 0 indicates no chunks")
       ->default_str(std::to_string(chunk_));
+    app->add_option("-r,--data-request-limit",
+                    "data request limit is pcie transfer width")
+      ->default_str(std::to_string(data_request_limit_))
+      ->check(CLI::IsMember(limits));
```
Comment (tswhison) [inline ofs/apps/cpeng/hps.cpp:110]: Isn't this going to restrict the value of --data-request-limit to one of the enumeration values, ie 0b00, 0b01, 0b10, 0b11?

Comment (r-rojo) [reply ofs/apps/cpeng/hps.cpp:110]: yes

File: ofs/apps/cpeng/hps.cpp [hunk 2]
```diff
@@ -45,11 +44,19 @@ constexpr size_t MB(uint32_t count) {
...
 
-constexpr size_t cacheline_aligned(size_t n) {
-  return (n + CACHELINE_SZ-1) & ~(CACHELINE_SZ-1);
+constexpr size_t aligned(size_t n, size_t sz) {
+  return (n + sz-1) & ~(sz-1);
 }
 const usec default_timeout_usec(msec(1000));
 
+// TODO: make enums in ofs cpeng yaml spec
+std::map<uint32_t, uint8_t> limit_map {
```
Comment (tswhison) [inline ofs/apps/cpeng/hps.cpp:53]: 👍

File: ofs/apps/cpeng/hps.cpp [hunk 3]
```diff
@@ -92,6 +101,10 @@ class cpeng : public opae::afu_test::command
       ->transform(CLI::AsNumberWithUnit(units));
     app->add_option("-c,--chunk", chunk_, "Chunk size. 0 indicates no chunks")
       ->default_str(std::to_string(chunk_));
+    app->add_option("-r,--data-request-limit",
```
Comment (tswhison) [inline ofs/apps/cpeng/hps.cpp:104]: Do you need to bind data_request_limit_ to the command option here?

Comment (r-rojo) [reply ofs/apps/cpeng/hps.cpp:104]: Yes, will fix

**Review by tswhison — APPROVED** [review]

**Pull request merged**